Source/Relativity.Test.Helpers/Relativity.Test.Helpers.NUnit.Integration/Kepler/OAuth2HelperTests.cs

```diff
@@ -0,0 +1,81 @@
+﻿using kCura.Relativity.Client;
+using NUnit.Framework;
+using Relativity.API;
+using Relativity.Services.Security;
+using Relativity.Test.Helpers.Kepler;
+using Relativity.Test.Helpers.SharedTestHelpers;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Relativity.Test.Helpers.NUnit.Integration.Kepler
+{
+	[TestFixture]
+	public class OAuth2HelperTests
+	{
+		private TestHelper _testHelper;
+		private string oAuthName = "TestOAuth2";
+		private OAuth2Helper Sut;
+
+		[OneTimeSetUp]
+		public void SetUp()
+		{
+			_testHelper = new TestHelper(TestContext.CurrentContext);
+
+			IOAuth2ClientManager oAuth2ClientManager = _testHelper.GetServicesManager().CreateProxy<IOAuth2ClientManager>(ExecutionIdentity.System);
+			IRSAPIClient rsapiClient = _testHelper.GetServicesManager().CreateProxy<IRSAPIClient>(ExecutionIdentity.System);
+
+			Sut = new OAuth2Helper(oAuth2ClientManager, rsapiClient);
+		}
+
+		[OneTimeTearDown]
+		public void TearDown()
+		{
+			_testHelper = null;
+			Sut = null;
+		}
+
+		[Test]
+		public async Task CreateOAuth2ClientAsyncTest()
+		{
+			// Arrange
+
+			// Act
+			Services.Security.Models.OAuth2Client oAuth2Client = await Sut.CreateOAuth2ClientAsync(ConfigurationHelper.ADMIN_USERNAME, oAuthName);
+
+			// Assert
+			Assert.IsTrue(oAuth2Client.Name.Equals(oAuthName));
+			await Sut.DeleteOAuth2ClientAsync(oAuth2Client.Id);

```

Comment:
I think it would be good if we can put the Delete code in a finally statement. 
Also before doing the delete, do a query to see if it exists and then proceed with the delete.

Reply:
Updated please review